File: CodeEval/medium/074minimumcoins/solution.rb
#!/usr/bin/env ruby

denom = [5, 3, 1]

File.open(ARGV[0]).each_line do |line|
  amt = line.to_i
  coins = 0
  denom.each do |n|
    qty = (amt / n).to_i
    amt -= qty * n
    coins += qty
  end
  puts coins
end

File: CodeEval/medium/074minimumcoins/solution.py
#!/usr/bin/env python

from sys import argv

DENOM = [5, 3, 1]

for x in open(argv[1]):
    amt = int(x)
    coins = 0
    for d in DENOM:
        qty = int(amt / d)
        amt -= qty * d
        coins += qty
    print(coins)
